### PR TITLE
Seller verification

### PR DESCRIPTION
**Pull request by kesslerjohn**
## New functionality
This branch allows admins to review and approve every seller's request to open a storefront. This feature brings more security and reliability to the Shoppr community, and incentivizes sellers to establish a positive relationship with their buyers.

## Examples
After a user registers as a seller, they will see the option to request a storefront in the user dropdown menu:
req_dropdown

They will be taken to a page where they can read about the review process for sellers creating a storefront (in this example, the user's name is "No Storefront" - the dropdown label still displays user name):
request_page

If they check the boxes and hit "Request", then an admin will receive an email with a link to review all pending storefront requests:
admin_receive_email

While the user is waiting for a response, their request status is pending. During this time, they cannot request to open another storefront:
status_pending

When an admin follows the email link, they will be taken to a page where they can view the profiles of sellers who have requested a storefront, and accept or reject the requests:
reqs_page

After approving a request, they are taken back to the requests page to continue reviewing requests, if any are left.

When a user's request is approved, they receive an email to let them know:
approved_email

and the next time they visit the site, they will have the "create storefront" option in their user dropdown menu (again, this is user "No Storefront"):
approved_dropdown
 ## Sad paths
A user who's rating is less than 3 stars or who has fewer than five reviews will not be able to request a storefront at all:
rating_too_low

not_enough_reviews

A user who is not logged in will not be able to view storefront requests:
admin_link_while_logged_out

And similarly, a user who is not an admin will not be able to view storefront requests. 

## To test
Create two users who are registered as sellers using an email you have access to - you can use the same email address if you run `bin/rake db:reset` and then `bin/rake db:seed` in between tests. When logged in as each of these sellers, request a storefront. 
Check the admin email address shopprwebsite@gmail.com to see that two emails have come in about requests. Log in as an admin on the Shoppr website, and then follow the link in the email to see the requests home page. 
Approve one seller and reject the other. Check the email for these users to see the response messages. Log into the app as either of these users and verify that you either can create a storefront, or cannot even request a storefront. 

## Specs
Coverage report generated for RSpec to /Users/jkessler/Desktop/3_FA_23/selt/SELT_REPOS/selt2023-final-project-team-001/coverage. 2240 / 2428 LOC (92.26%) covered.

### CODE REVIEW

**Comment by kesslerjohn** [comment]
Oh I forgot to add to the description but this is the email a user receives if their request is rejected:
Screen Shot 2023-12-10 at 2 10 25 AM

**Comment by powersaudrey25** [comment]
```
NoMethodError: undefined method `storefront_requesteds' for User:Class
/Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds/development.rb:8:in `<top (required)>'
/Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `load'
/Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `<top (required)>'
Tasks: TOP => db:reset => db:setup => db:seed

```
reqs = User.storefront_requesteds.values
typo?

**Comment by blhohmann** [comment]
The home page may need an update, if a user has not been approved by admin to create a storefront, instead of create a storefront it should probably be a card that says like request storefront

**Comment by kesslerjohn** [comment]
> ```
> NoMethodError: undefined method `storefront_requesteds' for User:Class
> /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds/development.rb:8:in `<top (required)>'
> /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `load'
> /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `<top (required)>'
> Tasks: TOP => db:reset => db:setup => db:seed
> ```
> 
> reqs = User.storefront_requesteds.values typo?

Oops, that used to be an enumerable (so storefront_requesteds.values would get the possible values it could take) but now it's an int. Quick fix (I think)

**Comment by kesslerjohn** [comment]
> The home page may need an update, if a user has not been approved by admin to create a storefront, instead of create a storefront it should probably be a card that says like request storefront

Oh good point - I'll see if I can add that

**Comment by kesslerjohn** [comment]
> a8da99d

This is fixed in [a8da99d](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/a8da99d185ebf9bd49735267301f743c5e2592be)

I am having an issue with a nil redirect that I can't track down. When I try to request a storefront as a seller without enough reviews, I get a 500 error for trying to redirect to nil. I can't find the source yet - let me know if you see that behavior.

**Comment by kesslerjohn** [comment]
> > ```
> > NoMethodError: undefined method `storefront_requesteds' for User:Class
> > /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds/development.rb:8:in `<top (required)>'
> > /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `load'
> > /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `<top (required)>'
> > Tasks: TOP => db:reset => db:setup => db:seed
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > reqs = User.storefront_requesteds.values typo?
> 
> Oops, that used to be an enumerable (so storefront_requesteds.values would get the possible values it could take) but now it's an int. Quick fix (I think)

This is fixed in [a8da99d](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/a8da99d185ebf9bd49735267301f743c5e2592be)

I am having an issue with a nil redirect that I can't track down. When I try to request a storefront as a seller without enough reviews, I get a 500 error for trying to redirect to nil. I can't find the source yet - let me know if you see that behavior.

**Comment by kesslerjohn** [comment]
> > > ```
> > > NoMethodError: undefined method `storefront_requesteds' for User:Class
> > > /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds/development.rb:8:in `<top (required)>'
> > > /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `load'
> > > /Users/audreypowers/SELT/selt2023-final-project-team-001/db/seeds.rb:105:in `<top (required)>'
> > > Tasks: TOP => db:reset => db:setup => db:seed
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > reqs = User.storefront_requesteds.values typo?
> > 
> > 
> > Oops, that used to be an enumerable (so storefront_requesteds.values would get the possible values it could take) but now it's an int. Quick fix (I think)
> 
> This is fixed in [a8da99d](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/a8da99d185ebf9bd49735267301f743c5e2592be)
> 
> I am having an issue with a nil redirect that I can't track down. When I try to request a storefront as a seller without enough reviews, I get a 500 error for trying to redirect to nil. I can't find the source yet - let me know if you see that behavior.

Redirect fixed in [946e638](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/946e6385dd33522cfef4508a52bd152d9924702e)

File: app/views/storefronts/make_request.html.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<h1>Request to make a storefront</h1>
```
Comment (powersaudrey25) [inline app/views/storefronts/make_request.html.erb:1]: redundant you could remove this line
```suggestion

```

**Comment by kesslerjohn** [comment]
> ec9cae5

Added in [ec9cae5](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/ec9cae5b058b7864b1905d405440430d561d5908)

**Comment by kesslerjohn** [comment]
> The home page may need an update, if a user has not been approved by admin to create a storefront, instead of create a storefront it should probably be a card that says like request storefront

Added in [ec9cae5](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/ec9cae5b058b7864b1905d405440430d561d5908)

**Comment by powersaudrey25** [comment]
> > The home page may need an update, if a user has not been approved by admin to create a storefront, instead of create a storefront it should probably be a card that says like request storefront
> 
> Added in [ec9cae5](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/ec9cae5b058b7864b1905d405440430d561d5908)
'Request Storefront' still links to create new storefront

**Comment by powersaudrey25** [comment]
One more thing, can we add the 'Show Requests' as an option for the drop down for admins? Also the 'Request Storefront' on the card still takes you to create new storefront not the request page...other than that looks good!

**Comment by kesslerjohn** [comment]
> One more thing, can we add the 'Show Requests' as an option for the drop down for admins? Also the 'Request Storefront' on the card still takes you to create new storefront not the request page...other than that looks good!

added in [12de5d8](https://github.com/uiowahjmjohnsonselt2023/selt2023-final-project-team-001/pull/69/commits/12de5d8a237a281a7cb934ef28e4291d7b9beb42)

**Review by powersaudrey25 — APPROVED** [review]

**Pull request merged**